syntax = "proto3";
package request_proto;

service Regulator {
    rpc GetTaskResult (stream TaskRequest) returns (stream TaskResponse) {}
}

message TaskRequest {
    bytes taskhash = 1;
}

message Result {
    bool status = 1;
    bytes task_result = 2; 
    string error_info = 3; // empty if status is true
}

message TaskResponse {
    bytes taskhash = 1;  
    Result result =2;  
    bytes rule_file_hash = 3;
    bytes signature = 4;
}